Tests/NIOTests/CircularBufferTests.swift

```diff
@@ -125,12 +136,30 @@ class CircularBufferTests: XCTestCase {
         XCTAssertEqual(ring.endIndex, 5)
 
         XCTAssertEqual(ring.index(after: 1), 2)
-
+        XCTAssertEqual(ring.index(before: 3), 2)
+        
         let actualValues = [Int](ring)
         let expectedValues = [0, 1, 2, 3, 4]
         XCTAssertEqual(expectedValues, actualValues)
     }
 
+    func testReplacingSubrangeOfElements() {
+        var ring = CircularBuffer<Int>(initialRingCapacity: 4)
+        for idx in 0..<50 {
+            ring.prepend(idx)
+        }
+        XCTAssertEqual(50, ring.count)
+        let newElements = ContiguousArray<Int>(arrayLiteral: 99)
```

Comment:
You shouldn't use the `arrayLiteral` initialiser directly, instead just do `let newElements: ContiguousArray<Int> = [99]`

Reply:
And in fact given that you're using `replaceSubrange` in this test you can just use a regular array.